src/porepy/compositional/states.py

```diff
@@ -0,0 +1,364 @@
+       residual, these data structures are used.
+
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Optional, Sequence, cast
+
+import numpy as np
+
+from .utils import (
+    compute_saturations,
+    extend_fractional_derivatives,
+    normalize_rows,
+    safe_sum,
+)
+
+__all__ = [
+    "ExtensiveState",
+    "IntensiveState",
+    "PhaseState",
+    "FluidState",
+    "initialize_fluid_state",
+]
+
+
+@dataclass
+class IntensiveState:
+    """Dataclass for storing intensive thermodynamic properties of a fluid mixture.
+
+    Storage is intended to be in array format for usage in flow and transport, where
+    the vectorized format reflects the degrees of freedom on a grid.
+
+    """
+
+    p: np.ndarray = field(default_factory=lambda: np.zeros(0))
+    """Pressure values. As of now, an equal pressure across all phases is assumed at
+    equilibrium."""
+
+    T: np.ndarray = field(default_factory=lambda: np.zeros(0))
+    """Temperature values. As of now, an equal temperatue across all phases is assumed
+    at equilibrium."""
+
+    z: np.ndarray = field(default_factory=lambda: np.zeros((0, 0)))
+    """Overall molar fractions (feed fractions) per component, stored in a 2D array
+    with row-wise values per component."""
+
+
+@dataclass
+class ExtensiveState:
+    """Dataclass for storing extensive thermodynamic properties
+    (analogous to :class:`IntensiveState`).
+
+    As of now, the extensive state does not encompass all of the physical quantities,
+    but those which are relevant for the flow & transport model in PorePy.
+
+    """
+
+    h: np.ndarray = field(default_factory=lambda: np.zeros(0))
+    """Specific enthalpy."""
+
+    rho: np.ndarray = field(default_factory=lambda: np.zeros(0))
+    """Density."""
+
+    @property
+    def v(self) -> np.ndarray:
+        """Specific volume as the reciprocal of :attr:`rho`.
+
+        Returns zeros, where :attr:`rho` is zero.
+
+        """
+        v = np.zeros_like(self.rho)
+        # special treatment for zero values to avoid division-by zero errors
+        idx = self.rho > 0.0
+        v[idx] = 1.0 / self.rho[idx]
+        return v
+
+
+@dataclass
+class PhaseState(ExtensiveState):
+    """An extended state description for physical phases, including derivatives
+    of extensive properties and properties which are not state functions such
+    as viscosity and thermal conductivity.
+
+    Derivatives are denoted with a ``d*`` and the derivative values are stored
+    along the first axis of a 2D array (i.e. each rot represents the derivative with
+    respect to 1 dependency)
+
+    The state of a phase is additionally characterized by an integer representing the
+    phase type, and the values of fractions per component.
+
+    """
+
+    phasetype: int = 0
+    """Type of the phase. Defaults to 0 (liquid)."""
+
+    x: np.ndarray = field(default_factory=lambda: np.zeros((0, 0)))
```

Comment:
I figured that.
That's why I use this naming scheme solely for the data structures in ``states.py``
``base.py`` and the mixins use PorePy's convention, which I understand.

But the code at the intersection points between mixins and the data structures would become unreadable, since the 88 character limit would render some very  obscure code.